packages/contracts-bedrock/src/L2/ETHLiquidity.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { WETH98 } from "src/dispute/weth/WETH98.sol";
+import { Predeploys } from "src/libraries/Predeploys.sol";
+import { L1Block } from "src/L2/L1Block.sol";
+import { SafeSend } from "src/universal/SafeSend.sol";
+import { ISemver } from "src/universal/ISemver.sol";
+
+import "src/libraries/errors/CommonErrors.sol";
+
+/// @title ETHLiquidity
+/// @notice The ETHLiquidity contract allows other contracts to access ETH liquidity without
+///         needing to modify the EVM to generate new ETH.
+contract ETHLiquidity {
+    /// @notice Emitted when an address burns ETH liquidity.
+    event LiquidityBurned(address indexed caller, uint256 value);
+
+    /// @notice Emitted when an address mints ETH liquidity.
+    event LiquidityMinted(address indexed caller, uint256 value);
+
+    /// @notice Semantic version.
+    /// @custom:semver 1.0.0
+    string public constant version = "1.0.0";
+
+    /// @notice Throws if the caller is not authorized.
+    modifier onlyAuthorized() {
+        // Only the SuperchainWETH contract is authorized to call this contract for now. Other
+        // contracts may be allowed to call this contract in the future so we're using a generic
+        // name for the modifier.
+        if (msg.sender != Predeploys.SUPERCHAIN_WETH) revert Unauthorized();
+        _;
+    }
+
+    /// @notice Throws this chain is using a custom gas token.
+    modifier notCustomGasToken() {
+        if (L1Block(Predeploys.L1_BLOCK_ATTRIBUTES).isCustomGasToken()) revert NotCustomGasToken();
+        _;
+    }
+
+    /// @notice Allows an address to lock ETH liquidity into this contract.
+    function burn() external payable onlyAuthorized notCustomGasToken {
+        emit LiquidityBurned(msg.sender, msg.value);
+    }
+
+    /// @notice Allows an address to unlock ETH liquidity from this contract.
+    /// @param _amount The amount of liquidity to unlock.
+    function mint(uint256 _amount) external onlyAuthorized notCustomGasToken {
+        new SafeSend{ value: _amount }(payable(msg.sender));
```

Comment:
I think this is a fine pattern to follow, altho is reentrancy a risk?

Reply:
Nope, SafeSend doesn't execute code so no reentrancy.

Reply:
I meant given that we don't call untrusted contracts here, there is no reentrancy risk, so we don't technically need to do this

Reply:
Oh hmmm, guess that's true. I figured it was just easier so that we'd never have to consider reentrancy down the line. Easier to close these doors off than accidentally opening up some reentrancy risk.